### PR TITLE
[boost-locale] Replace VCPKG_TARGET_IS_WINDOWS in b2-options

### LINKED ISSUE #13749: [boost-locale:x64-mingw-static] build failure
**Host Environment**
- OS: Windows (Insider Build 20221)
- Compiler: x64-mingw-static

**To Reproduce**
`./vcpkg --triplet x64-mingw-static install boost-locale`

**Failure logs** 
-Cut and past the appropriate build messages from the console output.
> CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:141 (message):
>     Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v
>     Working Directory: C:/vcpkg/buildtrees/boost-locale/x64-mingw-static-dbg
>     See logs for more information:
>       C:\vcpkg\buildtrees\boost-locale\install-x64-mingw-static-dbg-out.log
> 
> Call Stack (most recent call first):
>   scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
>   scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
>   installed/x64-mingw-static/share/boost-build/boost-modular-build.cmake:133 (vcpkg_install_cmake)
>   installed/x64-mingw-static/share/boost-build/boost-modular-build.cmake:138 (unix_build)
>   ports/boost-locale/portfile.cmake:13 (boost_modular_build)
>   scripts/ports.cmake:79 (include)

-Please attach any additional failure logs mentioned in the console output.
https://pastebin.com/qn8EQZFJ

### PR DESCRIPTION
**Pull request by longnguyen2004**
**Describe the pull request**

- What does your PR fix? Fixes #13749 
When we're in `unix_build`, `VCPKG_TARGET_IS_WINDOWS` isn't available, so fallback onto `CMAKE_SYSTEM_NAME` to check for Windows
- Which triplets are supported/not supported? Have you updated the CI baseline?
All mingw triplets
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

**Comment by longnguyen2004** [comment]
@comedinha Please test this PR

**Comment by comedinha** [comment]
@longnguyen2004 Fixed with this PR.

**Comment by longnguyen2004** [comment]
@PhoebeHui icu port isn't supported on mingw yet, can I skip the feature test?

**Comment by PhoebeHui** [comment]
@longnguyen2004, sure, I removed the tag.

**Review by PhoebeHui — APPROVED** [review]

**Comment by strega-nil** [comment]
I don't understand this, so I've added requires:discussion

**Comment by strega-nil** [comment]
Leaving this until Robert comes back, since he understands the boost modular build helpers better than anyone else.

**Comment by strega-nil** [comment]
@longnguyen2004 hey, I did a different fix for this issue; could you look at it and see if it works for your usecase?

**Comment by longnguyen2004** [comment]
@comedinha Can you please test it for me? My llvm-mingw toolchain is having some problems.

File: ports/boost-locale/b2-options.cmake.in [hunk 1]
```diff
@@ -1,15 +1,10 @@
-if(VCPKG_TARGET_IS_WINDOWS)
-    if("icu" IN_LIST FEATURES)
-        set(BOOST_LOCALE_ICU on)
-    else()
-        set(BOOST_LOCALE_ICU off)
-    endif()
-
+# When we get into unix_build, VCPKG_TARGET_IS_WINDOWS doesn't work, fall back onto CMAKE_SYSTEM_NAME
```
Comment (strega-nil) [inline ports/boost-locale/b2-options.cmake.in:1]:
```suggestion
```

**Comment by longnguyen2004** [comment]
@strega-nil Thanks for your help!

**Comment by strega-nil** [comment]
Thanks @longnguyen2004 for bringing it up in the first place :)

**Pull request merged**